gotext.go

```diff
@@ -132,38 +158,63 @@ func GetLibrary() string {
 	return lib
 }
 
-// SetLibrary sets the root path for the loale directories and files to be used at package level.
+// SetLibrary sets the root path for the locale directories and files to be used at package level.
 // It reloads the corresponding Translation file.
 func SetLibrary(lib string) {
 	globalConfig.Lock()
 	globalConfig.library = lib
 	globalConfig.Unlock()
 
-	loadStorage(true)
+	loadLocales(true)
 }
 
-// GetStorage is the locale storage getter for the package configuration.
-func GetStorage() *Locale {
+func GetLocales() []*Locale {
 	globalConfig.RLock()
-	storage := globalConfig.storage
-	globalConfig.RUnlock()
+	defer globalConfig.RUnlock()
+	return globalConfig.locales
+}
 
-	return storage
+// GetLocale is the locale storage getter for the package configuration.
+//
+// It returns the first locale returned by GetLocales.
+// This function exists for backwards compatibility, prefer using GetLocales directly.
+func GetLocale() *Locale {
+	locales := GetLocales()
+	if len(locales) > 0 {
+		return GetLocales()[0]
+	}
+	return nil
 }
 
-// SetStorage allows overridding the global Locale object with one built manually with NewLocale().
+// SetLocales allows for overriding the global Locale objects with ones built manually with
+// NewLocale(). This makes it possible to attach custom Domain objects from in-memory po/mo.
+// The library, language and domain of the first Locale will set the default global configuration.
+func SetLocales(locales []*Locale) {
+	globalConfig.Lock()
+	defer globalConfig.Unlock()
+
+	globalConfig.locales = locales
+	globalConfig.library = locales[0].path
+	globalConfig.domain = locales[0].defaultDomain
+
+	var languages []string
+	for _, locale := range locales {
+		languages = append(languages, locale.lang)
+	}
+	globalConfig.languages = languages
+}
+
+// SetLocale allows overriding the global Locale object with one built manually with NewLocale().
 // This allows then to attach to the locale Domains object in memory po or mo files (embedded or in any directory),
 // for each domain.
 // Locale library, language and domain properties will apply on default global configuration.
 // Any domain not loaded yet will use to the just in time domain loading process.
 // Note that any call to gotext.Set* or Configure will invalidate this override.
-func SetStorage(storage *Locale) {
```

Comment:
We can't remove support for this function. Lets make it a proxy to the new `SetLocale` instead and mark it for deprecation in `v2` (if v2 is ever happening)